src/transformers/modeling_utils.py

```diff
@@ -857,9 +858,14 @@ def _load_state_dict_into_meta_model(
 
     is_torch_e4m3fn_available = hasattr(torch, "float8_e4m3fn")
 
+    # We add this because HQQLinear dict has a very large state_dict (19 params/per module), which makes loading extremely slow
+    run_expected_keys_check = True
+    if isinstance(hf_quantizer, HqqHfQuantizer):
+        run_expected_keys_check = False
+
```

Comment:
with the comment below, this might not be needed

Reply:
Unfortunately we still need that check otherwise it breaks 
```Python
File ~/py310_env2/lib/python3.10/site-packages/accelerate/utils/modeling.py:358, in set_module_tensor_to_device(module, tensor_name, device, value, dtype, fp1
6_statistics, tied_params_map)
    355     return
    357 if old_value.device == torch.device("meta") and device not in ["meta", torch.device("meta")] and value is None:
--> 358     raise ValueError(f"{tensor_name} is on the meta device, we need a `value` to put in on {device}.")
    360 if value is not None:
    361     if old_value.shape != value.shape:

ValueError: weight is on the meta device, we need a `value` to put in on cuda:0
```

Reply:
Could you explain what happens here ? I don't understand why we are getting this error if we don't set that

Reply:
Answer here: https://github.com/huggingface/transformers/pull/33141/#issuecomment-2314984385

Reply:
Can you change the comment as it is not what's happening ?

Reply:
Resolved, see https://github.com/huggingface/transformers/pull/33141#issuecomment-2317474476 for the new version